packages/vscode/syntaxes/astro.tmLanguage.json

```diff
@@ -356,6 +359,169 @@
         { "include": "#html:tag-attributes" }
       ]
     },
+    "html:events": {
```

Comment:
Everything here might seems very scary, but it's mostly copied from VS Code's existing HTML definition that haven't been updated in two years. So it's relatively battle tested